src/values/acceleration-structure.cxx

```diff
@@ -0,0 +1,1484 @@
+        rayFlagOpaque = rayFlags & spv::RayFlagsMask::RayFlagsOpaqueKHRMask;
+        rayFlagNoOpaque = rayFlags & spv::RayFlagsMask::RayFlagsNoOpaqueKHRMask;
+        rayFlagTerminateOnFirstHit = rayFlags & spv::RayFlagsMask::RayFlagsTerminateOnFirstHitKHRMask;
+        rayFlagSkipClosestHitShader = rayFlags & spv::RayFlagsMask::RayFlagsSkipClosestHitShaderKHRMask;
+        rayFlagCullBackFacingTriangles = rayFlags & spv::RayFlagsMask::RayFlagsCullBackFacingTrianglesKHRMask;
+        rayFlagCullFrontFacingTriangles = rayFlags & spv::RayFlagsMask::RayFlagsCullFrontFacingTrianglesKHRMask;
+        rayFlagCullOpaque = rayFlags & spv::RayFlagsMask::RayFlagsCullOpaqueKHRMask;
+        rayFlagCullNoOpaque = rayFlags & spv::RayFlagsMask::RayFlagsCullNoOpaqueKHRMask;
+        rayFlagSkipTriangles = rayFlags & spv::RayFlagsMask::RayFlagsSkipTrianglesKHRMask;
+        rayFlagSkipAABBs = rayFlags & spv::RayFlagsMask::RayFlagsSkipAABBsKHRMask; // skip procedurals
+    }
+
+    void clearTrace() {
+        while (!nodesToEval.empty()) {
+            nodesToEval.pop();
+        }
+    }
+
+    void initTrace() {
+        closestInstance = nullptr;
+        closestPrimitiveID = 0;
+        committedIntersectionType = CommittedIntersectionType::None;
+        candidateIntersectionType = CandidateIntersectionType::Invalid;
+        activeTrace = true;
+        nodesToEval.push(&root);
+    }
+
+public: // TODO: also traversal stuff
+
+    void resetTrace() {
+        activeTrace = false;
+        clearTrace();
+        initTrace();
+    }
+    
+    void initTrace(const unsigned& rayFlags,
+            const unsigned& cullMask,
+            const std::vector<float>& rayOrigin,
+            const std::vector<float>& rayDirection,            
+            const float& rayTMin,
+            const float& rayTMax,
+            const bool& useSBT,
+            const unsigned offsetSBT = 0,
+            const unsigned strideSBT = 0,
+            const unsigned missIndex = 0) {
+        
+        assert(rayOrigin.size() == 3 && rayDirection.size() == 3);
+        glm::vec4 convertedRayOrigin = glm::make_vec4(rayOrigin.data());
+        convertedRayOrigin.w = 1.0f;
+        glm::vec4 convertedRayDirection = glm::make_vec4(rayDirection.data());
+        convertedRayDirection.w = 0.0f;
+
+        this->rayFlags = rayFlags;
+        this->cullMask = cullMask;
+        this->rayOrigin = convertedRayOrigin;
+        this->rayDirection = convertedRayDirection;
+        this->rayTMin = rayTMin;
+        this->rayTMax = rayTMax;
+
+        this->useSBT = useSBT;
+        this->offsetSBT = offsetSBT;
+        this->strideSBT = strideSBT;
+        this->missIndex = missIndex;
+
+        setFlags();
+        resetTrace();
+    }
+
+    void initTrace(const unsigned& rayFlags,
+            const unsigned& cullMask,
+            const glm::vec4& rayOrigin,
+            const glm::vec4& rayDirection,
```

Comment:
There is a significant amount of trailing spaces here (and elsewhere in the commit). You can do:

sed -i 's/[ \t]*$//' FILE_NAME

where you replace FILE_NAME with the name (if in the same directory), or you should be able to use the file path.

Reply:
Hopefully, I got all the files I modified